python/pyarrow/pandas-shim.pxi

```diff
@@ -174,6 +175,20 @@ cdef class _PandasAPIShim(object):
         self._check_import()
         return self._is_ge_v3
 
+    def is_ge_v3_strict(self):
+        self._check_import()
+        return self._is_ge_v3_strict
+
+    def uses_string_dtype(self):
+        if self.is_ge_v3_strict():
+            return True
+        try:
+            if self.pd.options.future.infer_string:
+                return True
+        except:
+            pass
+        return False
```

Comment:
Once pandas 3.0 is released this can be simplified, right? Should we open an issue so we don't forget to remove the interim env var and the check for it, which won't be necessary anymore

Reply:
It probably shouldn't be removed directly after 3.0 is released, because for quite a while this will still be useful for people install pandas 2.3 and enabling the option